lib/dao.js

```diff
@@ -1562,7 +1565,23 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
     var DataType = props[p] && props[p].type;
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
> Any test that queries on a nested property would cover this. There's already plenty of these.
  
None of them is failing with the current implementation though. Either add a test that fails with the current implementation, or simplify your changes by removing parts that are not needed to make newly added tests to pass.

See https://en.wikipedia.org/wiki/Test-driven_development#Test-driven_development_cycle